### DISCUSSION TITLE
M5Stick Plus + M5Stack Core2 w/ LVGL and BLE?

### DISCUSSION THREAD

**Discussion by efaden** [arendst/Tasmota#15068] [Q&A] [answered]
So I have a M5Stack Core2 and a M5Stick Plus.  I would like to run Tasmota on them to use BLE sensors (for now).  I am trying to figure out which firmware to install.  I tried installing the LVGL version, but that doesn't seem to have BLE.  Do I have to compile my own?  If so any suggestions on settings?

**Comment by Jason2866** [comment, marked as answer]
Add in `platformio_tasmota_cenv.ini`
```
[env:tasmota32-core2]
extends                 = env:tasmota32-lvgl
board                   = esp32-m5core2
build_flags             = ${env:tasmota32-lvgl.build_flags} -DUSE_I2S_SAY_TIME -DUSE_I2S_WEBRADIO -DUSE_SENDMAIL -DUSE_ESP32MAIL -DUSE_BLE_ESP32 -DUSE_MI_ESP32
lib_extra_dirs          = lib/libesp32, lib/libesp32_lvgl, lib/libesp32_div, lib/lib_basic, lib/lib_i2c, lib/lib_rf, lib/lib_div, lib/lib_ssl, lib/lib_display, lib/lib_audio
```
and compile this env with `platformio run -e tasmota32-core2`

**Reply by efaden** [reply]
Did that.... took the bin onto my linux box and flashed with the following command....  Won't boot.  Just clicks and makes a high pitch whine.... thoughts?
`./esptool --chip esp32 --port /dev/ttyACM0 --baud 1500000 --before default_reset write_flash -z --flash_mode dio --flash_freq 80m --flash_size detect 0x1000 /root/tasmota32-core2.bin`
<img width="820" alt="Screen Shot 2022-03-08 at 1 23 26 PM" src="https://user-images.githubusercontent.com/12599300/157301454-d62b68a6-5a50-4501-adec-3c2fdbdfc8a9.png">

**Comment by Jason2866** [comment]
@efaden  This is not correct. ESP32 needs more files for flashing. We have combined them in the `tasmota32-core2.factory.bin` which needs to be flashed at address `0x0`
So `./esptool --chip esp32 --port /dev/ttyACM0 --baud 1500000 --before default_reset write_flash -z --flash_mode dout  --flash_size detect 0x0 /root/tasmota32-core2.factory.bin`

**Reply by efaden** [reply]
OMG thank you!

**Reply by efaden** [reply]
@Jason2866 would the same build work on the M5StickC Plus?

**Reply by Jason2866** [reply]
i see no reason why not if PSRAM is there. LVGL need much RAM

**Reply by efaden** [reply]
Do I need to do anything special to flash that one?  Or just the normal bin?

**Reply by efaden** [reply]
@Jason2866 

Ran `esptool.py --chip esp32 --port /dev/myPort --baud 1500000 --before default_reset write_flash -z --flash_mode dout --flash_size detect 0x0 tasmota32-core2.factory.bin` on the M5StickC Plus.... Writes successfully.... but doesn't boot up.
